### PR TITLE
Minor Fluff Item Mechanics Tweak

### PR DESCRIPTION
**Pull request by EJDenton**
Changes the mechanics of the Display Stand and Spear to make it so you can remove the spear by simply clicking on the stand wherever it is and not holding the stand in your hand.

### CODE REVIEW

**Comment by EJDenton** [comment]
!review

**Review by Alberyk — APPROVED** [review]

File: code/modules/customitems/item_defines.dm [hunk 1]
```diff
@@ -654,6 +655,24 @@ All custom items with worn sprites must follow the contained sprite system: http
 	else
 		..()
 
+/obj/item/fluff/tokash_spear/MouseDrop(mob/user as mob)
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:658]:
```suggestion
/obj/item/fluff/tokash_spear/MouseDrop(mob/user)
```

File: code/modules/customitems/item_defines.dm [hunk 2]
```diff
@@ -654,6 +655,24 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+	if((user == usr && (!( usr.restrained() ) && (!( usr.stat ) && (usr.contents.Find(src) || in_range(src, usr))))))
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:659]:
```suggestion
	if((user == usr && (!(usr.restrained()) && (!(usr.stat) && (usr.contents.Find(src) || in_range(src, usr))))))
```
why are you checking if `user == usr` anyways?

File: code/modules/customitems/item_defines.dm [hunk 3]
```diff
@@ -654,6 +655,24 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+		if(!istype(usr, /mob/living/carbon/slime) && !istype(usr, /mob/living/simple_animal))
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:660]: kinda niche checks. check if the mob is human (i.e. a character) instead.

File: code/modules/customitems/item_defines.dm [hunk 4]
```diff
@@ -654,6 +655,24 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+			if( !usr.get_active_hand() )		//if active hand is empty
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:661]:
```suggestion
			if(!usr.get_active_hand()) // If active hand is empty.
```

File: code/modules/customitems/item_defines.dm [hunk 5]
```diff
@@ -636,7 +636,8 @@ All custom items with worn sprites must follow the contained sprite system: http
 	else
 		icon_state = "stand"
 
-/obj/item/fluff/tokash_spear/attack_self(var/mob/user)
+
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:639]:
```suggestion
```

File: code/modules/customitems/item_defines.dm [hunk 6]
```diff
@@ -636,7 +636,8 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+/obj/item/fluff/tokash_spear/attack_hand(var/mob/user)
 	if(has_spear)
 		to_chat(user, "<span class='notice'>You remove the spearhead from \the [src].</span>")
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:642]:
```suggestion
		to_chat(user, SPAN_NOTICE("You remove the spearhead from \the [src]."))
```

File: code/modules/customitems/item_defines.dm [hunk 7]
```diff
@@ -654,6 +655,24 @@ All custom items with worn sprites must follow the contained sprite system: http
...
 		..()
 
+/obj/item/fluff/tokash_spear/MouseDrop(mob/user as mob)
+	if((user == usr && (!( usr.restrained() ) && (!( usr.stat ) && (usr.contents.Find(src) || in_range(src, usr))))))
+		if(!istype(usr, /mob/living/carbon/slime) && !istype(usr, /mob/living/simple_animal))
+			if( !usr.get_active_hand() )		//if active hand is empty
+				var/mob/living/carbon/human/H = user
+				var/obj/item/organ/external/temp = H.organs_by_name[BP_R_HAND]
+
+				if (H.hand)
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:665]:
```suggestion
				if(H.hand)
```

File: code/modules/customitems/item_defines.dm [hunk 8]
```diff
@@ -654,6 +655,24 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+					temp = H.organs_by_name[BP_L_HAND]
+				if(temp && !temp.is_usable())
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:667]: i think there's a proc for this. (`use_check_and_message()`)

File: code/modules/customitems/item_defines.dm [hunk 9]
```diff
@@ -654,6 +655,24 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+					to_chat(user, "<span class='notice'>You try to move your [temp.name], but cannot!</span>")
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:668]:
```suggestion
					to_chat(user, SPAN_NOTICE("You try to move your [temp.name], but cannot!"))
```

File: code/modules/customitems/item_defines.dm [hunk 10]
```diff
@@ -654,6 +655,24 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+					return
+
+				to_chat(user, "<span class='notice'>You pick up the [src].</span>")
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:671]:
```suggestion
				to_chat(user, SPAN_NOTICE("You pick up \the [src]."))
```

File: code/modules/customitems/item_defines.dm [hunk 11]
```diff
@@ -654,6 +655,24 @@ All custom items with worn sprites must follow the contained sprite system: http
...
+				user.put_in_hands(src)
+
```
Comment (SleepyGemmy) [inline code/modules/customitems/item_defines.dm:673]:
```suggestion
```

File: html/changelogs/Evandorf-FluffItemChange.yml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Very minor changes to fluff item mechanics. Can now remove spearhed from stand without needing to pick up the stand first."
```
Comment (SleepyGemmy) [inline html/changelogs/Evandorf-FluffItemChange.yml:41]:
```suggestion
  - tweak: "Very minor changes to fluff item mechanics. Can now remove spearhead from stand without needing to pick up the stand first."
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

File: code/modules/customitems/item_defines.dm (hunk 3 above)
Comment (EJDenton) [reply code/modules/customitems/item_defines.dm:659]: I pulled this MouseDrop code from  the paperbin code.
Aurora.3/code/modules/paperwork/paperbin.dm

I'm not sure how I would rewrite it to make a different chcek.
Possibly if(istype(usr, /mob/living/carbon/human) ?

File: code/modules/customitems/item_defines.dm (hunk 8 above)
Comment (EJDenton) [reply code/modules/customitems/item_defines.dm:666]: I'm not entire sure what you mean by this. Are you saying that 660 through 666 are unnecessary or that a different line using use_check_and_message() should be used?

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**